tests/hps/utils.py

```diff
@@ -0,0 +1,17 @@
+import collections
+from typing import Any, Dict, List, Union
+
+from embeddings.pipeline.pipelines_metadata import LightningPipelineMetadata
+
+
+def _flatten(
```

Comment:
Maybe move this utility to our library core?

Reply:
The mechanism of this function is quite primitive and can be dangerous. I would not recommend moving it.

Reply:
ok, I understand it